core/app/[locale]/(default)/(faceted)/category/[slug]/page.tsx

```diff
@@ -47,97 +68,184 @@ export async function generateMetadata({ params }: Props): Promise<Metadata> {
+    price: pricesTransformer(product.prices, format),
+    subtitle: product.brand?.name ?? undefined,
+  }));
+}
 
-  const productsCollection = search.products;
-  const products = productsCollection.items;
-  const { hasNextPage, hasPreviousPage, endCursor, startCursor } = productsCollection.pageInfo;
+async function getFilters(
+  categoryId: number,
+  searchParamsPromise: Promise<SearchParams>,
+): Promise<Filter[]> {
+  const searchParams = await searchParamsPromise;
+  const searchParamsCache = await createFiltersSearchParamCache(categoryId);
+  const parsedSearchParams = searchParamsCache.parse(searchParams);
+  const categorySearch = await fetchFacetedSearch({ category: categoryId });
+  const refinedSearch = await fetchFacetedSearch({
+    ...searchParams,
+    ...parsedSearchParams,
+    category: categoryId,
+  });
+  const allFacets = categorySearch.facets.items.filter(
+    (facet) => facet.__typename !== 'CategorySearchFilter',
+  );
+  const refinedFacets = refinedSearch.facets.items.filter(
+    (facet) => facet.__typename !== 'CategorySearchFilter',
+  );
+
+  const transformedFacets = await facetsTransformer({
+    refinedFacets,
+    allFacets,
+    searchParams: { ...searchParams, ...parsedSearchParams },
+  });
+
+  return transformedFacets.filter((facet) => facet != null);
+}
+
+async function getSortLabel(): Promise<string> {
+  const t = await getTranslations('FacetedGroup.SortBy');
+
+  return t('ariaLabel');
+}
+
+async function getSortOptions(): Promise<SortOption[]> {
+  const t = await getTranslations('FacetedGroup.SortBy');
+
+  return [
+    { value: 'featured', label: t('featuredItems') },
+    { value: 'newest', label: t('newestItems') },
+    { value: 'best_selling', label: t('bestSellingItems') },
+    { value: 'a_to_z', label: t('aToZ') },
+    { value: 'z_to_a', label: t('zToA') },
+    { value: 'best_reviewed', label: t('byReview') },
+    { value: 'lowest_price', label: t('priceAscending') },
+    { value: 'highest_price', label: t('priceDescending') },
+    { value: 'relevance', label: t('relevance') },
+  ];
+}
+
+async function getPaginationInfo(
+  categoryId: number,
+  searchParamsPromise: Promise<SearchParams>,
+): Promise<CursorPaginationInfo | null> {
+  const searchParams = await searchParamsPromise;
+  const searchParamsCache = await createFiltersSearchParamCache(categoryId);
+  const parsedSearchParams = searchParamsCache.parse(searchParams);
+  const search = await fetchFacetedSearch({
+    ...searchParams,
+    ...parsedSearchParams,
+    category: categoryId,
+  });
+  const { hasNextPage, hasPreviousPage, endCursor, startCursor } = search.products.pageInfo;
+
+  return hasNextPage || hasPreviousPage
+    ? {
+        startCursorParamName: 'before',
+        endCursorParamName: 'after',
+        endCursor: hasNextPage ? endCursor : null,
+        startCursor: hasPreviousPage ? startCursor : null,
+      }
+    : null;
+}
+
+async function getEmptyStateTitle(): Promise<string | null> {
```

Comment:
We have to do this in other components, but I dislike it 😢  haha.

Reply:
Yeah, it's unfortunate. I think that once we've got a better way of doing localization in VIBES, this will go away since we'll just provide a dictionary of labels all centralized in one place